services/frontend-service/src/ui/AppBar/LocksDrawer.tsx

```diff
@@ -0,0 +1,171 @@
+                                noWrap
+                                variant="h5"
+                                component="div"
+                                className="name"
+                                width="40%"
+                                sx={{ textTransform: 'capitalize' }}>
+                                {lock.type}
+                            </Typography>
+                            <Typography
+                                noWrap
+                                variant="h5"
+                                component="div"
+                                className="name"
+                                width="50%"
+                                sx={{ textTransform: 'capitalize' }}>
+                                {lock.message}
+                            </Typography>
+                        </Paper>
+                    </Grid>
+                    <Divider />
+                </>
+            ))}
+        </List>
+    );
+};
+
+export const LocksDrawer = (props: { data: GetOverviewResponse }) => {
+    const { data } = props;
+    const [state, setState] = React.useState({ isOpen: false });
+    const [locks, setLocks] = React.useState<ILocks[]>([]);
+    const [outDatedLocks, setOutDatedLocks] = React.useState<boolean>(false);
+
+    const toggleDrawer = (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
+        if (
+            event.type === 'keydown' &&
+            ((event as React.KeyboardEvent).key === 'Tab' || (event as React.KeyboardEvent).key === 'Shift')
+        ) {
+            return;
+        }
+        setState({ isOpen: open });
+    };
+
+    const calcAge = useCallback((time: any) => {
+        if (!time) return -1;
+        const curTime = new Date().getTime();
+        const diffTime = curTime - time;
+        const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+        return diffDays;
+    }, []);
+
+    useEffect(() => {
+        let nwLocks: ILocks[] = [];
+        for (const env of Object.values(data.environments)) {
+            nwLocks = [
+                ...nwLocks,
+                ...Object.keys(env.locks ?? {}).map((value) => ({
+                    id: value,
+                    ...env.locks[value],
+                    type: env.name,
+                    age: calcAge(env.locks[value].commit?.authorTime),
+                })),
+            ];
+
+            for (const app of Object.values(env.applications)) {
+                nwLocks = [
+                    ...nwLocks,
+                    ...Object.keys(app.locks ?? {}).map((value) => ({
+                        id: value,
+                        ...app.locks[value],
+                        type: app.name,
+                        age: calcAge(app.locks[value].commit?.authorTime),
+                    })),
+                ];
+            }
+        }
+        nwLocks.sort((a: ILocks, b: ILocks) => {
+            if (!a.commit?.authorTime || !b.commit?.authorTime) return 0;
```

Comment:
why is the `!` necessary? We don't have null locks, or do we?

Reply:
Not the lock but the commit and the authorTime can be null

Reply:
I think in that case you can remove the `!`. (unless the compiler complains)

Reply:
Yes, the compiler is complaining about it

Reply:
huh, weird, but ok, then keep it.